### PR TITLE
cli/pack: erase temp file on failure

### PR DESCRIPTION
**Pull request by pleshevskiy**
Moved path validation above. Added zip archive cleanup on encryption error.

I want to move password retrieval from encryption, but it will take more time and effort)

### CODE REVIEW

File: dexios/src/subcommands/hashing.rs [hunk 1]
```diff
@@ -7,7 +7,7 @@ use std::io::Read;
 // this hashes the input file
 // it reads it in blocks, updates the hasher, and finalises/displays the hash
 // it's used by hash-standalone mode
-pub fn hash_stream(files: &Vec<String>) -> Result<()> {
+pub fn hash_stream(files: &[String]) -> Result<()> {
```
Comment (brxken128) [inline dexios/src/subcommands/hashing.rs:10]: This is a good change - I don't know how I missed this

Comment (pleshevskiy) [reply dexios/src/subcommands/hashing.rs:10]: thanks `clippy` for finding this :)

**Comment by brxken128** [comment]
> Moved path validation above. Added zip archive cleanup on encryption error.
> 
> I want to move password retrieval from encryption, but it will take more time and effort)

I'm not too sure how good of an idea it would be to move password-handling from outside of the encryption function. I understand that it's one of the sources of possible errors (when a keyfile is supplied), but I'm not too sure we should move it out. If we did, we wouldn't be able to check for missing files beforehand.

One way around this could be to have a kind of "entry/validation function", something that globally checks for common mishaps. This function could check that the input file exists, checks that the keyfile exists/isn't empty, get the user's password, ask if they'd like to overwrite the output file, etc. It'd only be needed for some functions (not the header manipulation ones), but it could be an ideal way to handle this I think.

The other functions could then receive the user's key, in the form of a `Protected<Vec<u8>>` (and continue life as usual).

**Comment by pleshevskiy** [comment]
> If we did, we wouldn't be able to check for missing files beforehand.

I want to get the password after checking the files, but before creating the compressed file) I think this is correct) in fact for this reason I moved the walker with the file path checker to the top.

**Comment by pleshevskiy** [comment]
Just give me a chance to finish this task :)

**Comment by pleshevskiy** [comment]
@brxken128 actually, I think we should merge this PR ) My commit solves #99, but I want to continue in separate branch.

File: dexios/src/subcommands/pack.rs [hunk 2]
```diff
@@ -19,37 +19,57 @@ use crate::{
...
+        logger.info(format!("Traversing {}", req.input_file));
+        WalkDir::new(req.input_file).max_depth(1)
+    };
+
+    // 2. Skip failed dir entries
+    let walker = walker
+        .into_iter()
+        .filter_map(|res| res.ok())
+        .collect::<Vec<walkdir::DirEntry>>();
+    // let item_data = item.context("Unable to get path of item, skipping")?;
```
Comment (brxken128) [inline dexios/src/subcommands/pack.rs:53]: Dead code that could do with being removed :)

**Comment by brxken128** [comment]
I don't have the chance to test this this myself, but the tests pass and the code looks great so I'll merge. Great work! @pleshevskiy

File: dexios/src/subcommands/pack.rs (hunk 2 above)
Comment (pleshevskiy) [reply dexios/src/subcommands/pack.rs:53]: I've pushed... but I was too late

Comment (brxken128) [reply dexios/src/subcommands/pack.rs:53]: No worries, I'll remove it in a sec :)

**Pull request merged**